src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlDictionaryAsyncCheckWriter.cs

```diff
@@ -694,5 +696,14 @@ protected override void Dispose(bool disposing)
             CheckAsync();
             CoreWriter.Dispose();
         }
+
+        public void SetOutput(Stream stream, Encoding encoding, bool ownsStream)
```

Comment:
Should any of the arguments be validated (e.g. for being non-null) even if there isn't an initializer?

Reply:
We would be double validating. I think we can let the inner implementation deal with validation. I don't believe that _initializer can ever be null as we only ever wrap something which implements IXmlTextWriterInitializer. Maybe we should have a Debug.Assert in the constructor to validate that expectation, or even throw in the constructor if we aren't wrapping an IXmlTextWriterInitializer.